### PR TITLE
Add `initContainers` and `extraContainers`

### LINKED ISSUE #16: Ability to add a initContainer and a sidecar to the server
In the case of using [litestream ](https://litestream.io/) when the server uses the sqlite database, we should be able to add a initContainer and a sidecar on the server component.

https://litestream.io/guides/kubernetes/

PS - I will do a PR in a few days.

### PR DESCRIPTION
**Pull request by babykart**
closes #16

### CODE REVIEW

**Review by pat-s — COMMENTED** [review]
The reference to litestream.io feels a bit like an (unneeded) plug (?).

**Comment by babykart** [comment]
@pat-s My first intention with this PR was to address a [litestream use case](https://github.com/woodpecker-ci/woodpecker/issues/909).
But effectively, the reference itself can be removed.
For the moment, I don't see another use case that needs the extraContainers, initContainers and extraVolumes but they could be usefull for some specific needs.

**Comment by pat-s** [comment]
@babykart Do you want to update this PR? `extraVolumes` has already been added meanwhile.

I would personally prefer to remove the `litestream` ref and just add dummy placeholders. Examples are always great, especially when the config is nested but the one in this PR is also a bit long. Maybe a link to an existing example repo/project would be better in this case to keep the values file tidy.

**Comment by babykart** [comment]
@pat-s No problem. I will update this PR ASAP.

**Comment by pat-s** [comment]
@babykart ping :)

**Comment by babykart** [comment]
@pat-s done !

PS - Sorry for the lag.

File: values.yaml [hunk 1]
```diff
@@ -195,6 +195,63 @@ server:
   # -- Overrides the full name of the helm chart of the server component
   fullnameOverride: ""
 
+  # -- Add extra containers to the server component
+  extraContainers: []
+  # Example from https://litestream.io/guides/kubernetes/#litestream-application-container
```
Comment (lafriks) [inline values.yaml:200]: Imho these examples should be put somewhere in the docs not here

Comment (babykart) [reply values.yaml:200]: @lafriks  This stuff has been removed in 0d9c83407d3a3d0bf8636f9e0aef3d7a69485308

Comment (lafriks) [reply values.yaml:200]: but some how it still shows up in PR diff 🤔

**Comment by pat-s** [comment]
@babykart NB: `extraVolumes` has been added in #78, so a merge is needed.

**Pull request closed without merging**